code/modules/stories_system/code/actor/ghost/ghost_actors.dm

```diff
@@ -0,0 +1,134 @@
+
+	var/obj/landmark
+	for(var/obj/effect/landmark/ghost_story_participant/actor_spawn in GLOB.landmarks_list)
+		if(actor_spawn.actor_id != actor_spawn_id)
+			continue
+		landmark = actor_spawn
+		break
+
+	// Handling for reserving a new area
+	if(!landmark)
+		var/datum/map_template/actor_spawn/new_actor = new
+		var/datum/turf_reservation/reservation_area = SSmapping.RequestBlockReservation(new_actor.width, new_actor.height, SSmapping.transit.z_value, /datum/turf_reservation/transit)
+		if(!reservation_area)
+			CRASH("failed to reserve an area for actor spawning")
+		var/turf/bottom_left = TURF_FROM_COORDS_LIST(reservation_area.bottom_left_coords)
+		new_actor.load(bottom_left, centered = FALSE)
+		var/list/affected = new_actor.get_affected_turfs(bottom_left, centered=FALSE)
+		for(var/turf/affected_turf as anything in affected)
+			for(var/obj/effect/landmark/ghost_story_participant/actor_spawn in affected_turf)
+				actor_spawn.actor_id = actor_spawn_id
+		for(var/obj/effect/landmark/ghost_story_participant/actor_spawn in GLOB.landmarks_list)
+			if(actor_spawn.actor_id != actor_spawn_id)
+				continue
+			landmark = actor_spawn
+			break
+		if(!landmark)
+			CRASH("failed to find an actor spawn ID landmark after making a template that should've assigned one!")
+
+	// Spawning the actor
+	var/mob/living/carbon/human/temporary_human = new(get_turf(landmark))
+	temporary_human.key = picked_spawner.key
+	temporary_human.client?.prefs?.safe_transfer_prefs_to(temporary_human)
+	temporary_human.equip_to_slot(new /obj/item/clothing/under/color/grey, ITEM_SLOT_ICLOTHING)
+	current_story.mind_actor_list[temporary_human.mind] = src
+	info_button = new(src)
+	info_button.Grant(temporary_human)
+	temporary_human.balloon_alert(temporary_human, "[delay_time / 600] minute(s) until you're sent in")
+	addtimer(CALLBACK(src, .proc/send_them_in, temporary_human), delay_time)
+
+/// Ghost ones are delayed by a few minutes to ensure that the ghosts have time to choose a character and for crew actors to get settled a bit
+/datum/story_actor/ghost/proc/send_them_in(mob/living/carbon/human/to_send_human)
+	to_send_human.equipOutfit(pick(actor_outfits))
+
+/datum/story_actor/ghost/centcom_inspector
+	name = "Central Command Inspector"
+	actor_outfits = list(
+		/datum/outfit/centcom_inspector,
+	)
+	actor_info = "You are an inspector from Central Command on a surprise inspection of the station. Inspect things, pick out issues, and make a good report for CentCom."
+	actor_goal = "Inspect the station, make a report, and send it to CentCom when done."
+
+/datum/story_actor/ghost/centcom_inspector/send_them_in(mob/living/carbon/human/to_send_human)
+	to_send_human.client?.prefs?.safe_transfer_prefs_to(to_send_human)
+	. = ..()
+	var/area/station/command/bridge/bridge_area = GLOB.areas_by_type[/area/station/command/bridge]
+	var/list/bridge_tiles = list()
+	for(var/turf/open/floor/floor_tile in bridge_area)
```

Comment:
Don't loop over areas. Use `get_area_contents()` or whatever it's called instead.